File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {

        int num = nums.length / 2;
        Set<Integer> hs = new HashSet<Integer>();
        for (int i = 0; i < nums.length; i++){
            hs.add(nums[i]);
        }

        int pokemon = hs.size();
        if (pokemon <= num){
            return pokemon;
        }else{
            return num;
        }
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
# def solution(nums):
#     cnt = set(nums)
#     if len(nums)/2 > len(cnt):
#         answer = len(cnt)
#     else:
#         answer = len(nums)/2
#     return answer

def solution(nums):
    return min([len(set(nums)),len(nums)/2])